datastore/longhorn.go

```diff
@@ -1767,7 +1771,7 @@ func (s *DataStore) ListNodesRO() ([]*longhorn.Node, error) {
 	return s.nLister.Nodes(s.namespace).List(labels.Everything())
 }
 
-func (s *DataStore) ListNodesWithEngineImage(image string) (map[string]*longhorn.Node, error) {
+func (s *DataStore) ListNodesWithEngineImage(image string, states map[types.EngineImageState]struct{}) (map[string]*longhorn.Node, error) {
```

Comment:
Just notice that this function looks weird. I think we can directly input an EngineImage CR and filter the CR outside of this function. This function is only responsible for picking up all nodes contain this EngineImage.